tests/integration/test_discourse.py

```diff
@@ -106,27 +101,20 @@ async def test_create_topic_auth_error(
 
 @pytest.mark.asyncio
 async def test_retrieve_topic_auth_error(
-    discourse_user_api_key: str,
     discourse_hostname: str,
     discourse_user_credentials: types.Credentials,
```

Comment:
Whilst it is possible, what would be the benefits? `discourse_user_credentials` is a pytest fixture that is automatically passed to the test.

The purpose of these series of tests is to check that interactions with discourse fail when expected, which is usually achieved by deliberately using incorrect credentials. See, for example, `test_read_write_permission` which takes an alternate credential. Taking the fixture as a parameter also makes it explicit that the test depends on that fixture rather than implicit vis getting it from `discourse_api`

Reply:
I'm fine with keeping it like this, I tend to try to minimize the signature size whenever I can, but we're in unit tests and using a fixture so that doesn't really add value